### PR TITLE
Spark: Remove deprecated AssertHelpers

### PR DESCRIPTION
**Pull request by liuxiaocs7**
- SubTask of https://github.com/apache/iceberg/issues/7094

Remove `AssertHelpers` in `iceberg-spark` module.

### CODE REVIEW

File: spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestWriteAborts.java [hunk 1]
```diff
@@ -105,23 +105,22 @@ public void testBatchAppend() throws Exception {
...
-                .coalesce(1)
-                .sortWithinPartitions("id")
-                .writeTo(tableName)
-                .option(SparkWriteOptions.USE_TABLE_DISTRIBUTION_AND_ORDERING, "false")
-                .append();
-          } catch (NoSuchTableException e) {
-            throw new RuntimeException(e);
-          }
-        });
+    Assertions.assertThatThrownBy(
```
Comment (nastra) [inline spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestWriteAborts.java:108]:
```suggestion
    Assertions.assertThatThrownBy(
            () ->
                // incoming records are not ordered by partitions so the job must fail
                inputDF
                    .coalesce(1)
                    .sortWithinPartitions("id")
                    .writeTo(tableName)
                    .option(SparkWriteOptions.USE_TABLE_DISTRIBUTION_AND_ORDERING, "false")
                    .append())
        .isInstanceOf(SparkException.class)
        .hasMessageContaining("Encountered records that belong to already closed files");
```

File: spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java [hunk 2]
```diff
@@ -69,21 +69,22 @@ public void testOverwriteFilterSerializableIsolation() throws Exception {
...
-                .option(SparkWriteOptions.VALIDATE_FROM_SNAPSHOT_ID, String.valueOf(snapshotId))
-                .option(SparkWriteOptions.ISOLATION_LEVEL, IsolationLevel.SERIALIZABLE.toString())
-                .overwrite(functions.col("id").equalTo(1));
-          } catch (NoSuchTableException e) {
-            throw new RuntimeException(e);
-          }
-        });
+    Assertions.assertThatThrownBy(
+            () -> {
+              try {
```
Comment (nastra) [inline spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java:74]: the try-catch can be removed here

File: spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java [hunk 3]
```diff
@@ -111,21 +112,21 @@ public void testOverwriteFilterSerializableIsolation2() throws Exception {
...
-                .option(SparkWriteOptions.VALIDATE_FROM_SNAPSHOT_ID, String.valueOf(snapshotId))
-                .option(SparkWriteOptions.ISOLATION_LEVEL, IsolationLevel.SNAPSHOT.toString())
-                .overwrite(functions.col("id").equalTo(1));
-          } catch (NoSuchTableException e) {
-            throw new RuntimeException(e);
-          }
-        });
+    Assertions.assertThatThrownBy(
+            () -> {
+              try {
```
Comment (nastra) [inline spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java:117]: same as above

File: spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java [hunk 4]
```diff
@@ -409,20 +414,21 @@ public void testOverwritePartitionNoSnapshotIdValidation() throws Exception {
...
-                .writeTo(tableName)
-                .option(SparkWriteOptions.ISOLATION_LEVEL, IsolationLevel.SERIALIZABLE.toString())
-                .overwritePartitions();
-          } catch (NoSuchTableException e) {
-            throw new RuntimeException(e);
-          }
-        });
+    Assertions.assertThatThrownBy(
+            () -> {
+              try {
```
Comment (nastra) [inline spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java:419]: we can remove all try-catch inside `assertThatThrownBy` blocks

Comment (liuxiaocs7) [reply spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestConflictValidation.java]: Thanks, already removed all~

**Review by nastra — APPROVED** [review]
LGTM, thanks @liuxiaocs7

**Pull request merged**